docs/usage/types/dicts_mapping.md

```diff
@@ -18,3 +61,69 @@
     Prior to Python 3.8, it requires the [typing-extensions](https://pypi.org/project/typing-extensions/) package.
     But required and optional fields are properly differentiated only since Python 3.9.
     We therefore recommend using [typing-extensions](https://pypi.org/project/typing-extensions/) with Python 3.8 as well.
+
+Same as `dict` but Pydantic will validate the dictionary since keys are annotated.
+You can define `__pydantic_config__` to change the model inherited from `TypedDict`.
+See [Model Config](../model_config.md) for more details.
+
+```py
+from typing import Optional
+
+from typing_extensions import TypedDict
+
+from pydantic import BaseModel, ConfigDict, ValidationError
```

Comment:
I think this should be split into multiple code examples - the simplest showing the very basic use of `TypedDict`, then showing more advanced usage like Config` and `total=False`

Reply:
Done